decidim-elections/app/controllers/decidim/votings/census/admin/census_controller.rb

```diff
@@ -84,40 +66,23 @@ def current_census
             @current_census ||= Dataset.find_by(
               organization: current_organization,
               voting: current_participatory_space
-            ) || Dataset.new(status: :create_data)
-          end
-
-          def census_steps
-            [
-              ["Create census", current_step(:create_data)],
-              ["Review Data", current_step(:review_data)],
-              ["Create Acess Codes", current_step(:generate_codes)],
-              ["Export Acess Codes", current_step(:export_codes)],
-              ["Unlock Election", current_step(:freeze)]
-            ]
-          end
-
-          def current_step(step)
-            if step == current_census.status.to_sym
-              :current
-            else
-              current_pointer = Dataset.statuses[current_census.status]
-              pointer = Dataset.statuses[step]
-
-              pointer > current_pointer ? :next : :prev
-            end
+            ) || Dataset.new(status: :init_data)
           end
 
           def admin_voting_census_path
             decidim_votings_admin.voting_census_path(current_participatory_space)
           end
 
+          # [:init_data, :creating_data, :data_created, :generating_codes, :codes_generated, :freeze]
+
           def current_census_action_view
             case current_census.status
-            when "create_data"
+            when "init_data"
```

Comment:
💬 we could use the enum methods to check for the different status to make it less error prone:
```ruby
if current_census.init_data?
  "new_census"
elsif current_census.creating_data?
  "creating_data"
....
```